src/lib/circle.js

```diff
@@ -0,0 +1,176 @@
+        return clipboardValue.length === 40 ? clipboardValue : null;
+      },
+      when: async () => {
+        try {
+          const storedToken = await passwordStorage.get('token');
+          return !info.options.keychain || info.options['ask-for-passwords'] || !storedToken;
+        } catch (err) {
+          info.log.error(
+            'Something went wrong with your stored api token. Delete them from your keychain and try again'
+          );
+          process.exit(1); // eslint-disable-line unicorn/no-process-exit
+        }
+      },
+    },
+    {
+      type: 'confirm',
+      name: 'createConfigFile',
+      message: 'Do you want a `config.yml` file with semantic-release setup?',
+      default: true,
+    },
+    {
+      // Add step to existing config.yml later
+      type: 'confirm',
+      name: 'overwrite',
+      default: false,
+      message: 'Do you want to overwrite the existing `config.yml`?',
+      when: answers => answers.createConfigFile && fs.existsSync('./.circleci/config.yml'),
+    },
+  ]);
+}
+
+async function processToken(info) {
+  if (!info.circle.token) {
+    info.circle.token = await passwordStorage.get('token');
+  } else if (info.options.keychain) {
+    passwordStorage.set('token', info.circle.token);
+  }
+}
+
+async function setupCircleProject(info) {
+  const defaultReq = request.defaults({
+    headers: {
+      'content-type': 'application/json',
+      accept: 'application/json',
+    },
+    qs: {
+      'circle-token': info.circle.token,
+    },
+    baseUrl: `https://circleci.com/api/v1.1/project/github/${info.ghrepo.slug[0]}/${info.ghrepo.slug[1]}/`,
+    json: true,
+    simple: true,
+  });
+
+  await followProject(info, defaultReq);
+  await addEnvironmentVariable(info, defaultReq, {name: 'GH_TOKEN', value: info.github.token});
+  await addEnvironmentVariable(info, defaultReq, {name: 'NPM_TOKEN', value: info.npm.token});
+}
+
+async function followProject(info, defaultReq) {
+  info.log.verbose(`Following repo ${info.ghrepo.slug[0]}/${info.ghrepo.slug[1]} on CircleCI...`);
+  const uri = `/follow`;
+  await defaultReq
+    .post(uri)
+    .then(() => {
+      info.log.info(`Succesfully followed repo ${info.ghrepo.slug[0]}/${info.ghrepo.slug[1]} on CircleCI.`);
+    })
+    .catch(() => {
+      info.log.error('Error following repo on CircleCI!');
+      process.exit(1); // eslint-disable-line unicorn/no-process-exit
+    });
+}
+
+async function addEnvironmentVariable(info, defaultReq, body) {
+  info.log.verbose(`Adding environment variable ${body.name} to CircleCI project...`);
+  const uri = `/envvar`;
+  await defaultReq
+    .post(uri, {body})
+    .then(() => {
+      info.log.info(`Successfully added environment variable ${body.name} to CircleCI project.`);
+    })
+    .catch(() => {
+      info.log.error('Error setting environment variables on CircleCI!');
+      process.exit(1); // eslint-disable-line unicorn/no-process-exit
+    });
+}
+
+function setupRequestLogging(info) {
```

Comment:
What is the purpose of this method? Can't just use the logger like in the rest of the code?

Is it used to hide the token in the log output?

Reply:
To have a central place to add logging. It's only for logging purposes. I'm still using `request` for the actual requests. This keeps my actual calls easier, because they return the body as JSON directly (because of `json: true` and `simple: true`, while the handler for `request-debug` still allows me to log status code, full uri, etc.) It's just a separation of concerns.

Reply:
ok